### PR TITLE
Add GPU device compute capability check for LTensor

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

Add GPU device compute capability check. L-Tensor has device compute capability check like L-GPU now.
[SC-69118]

**Description of the Change:**

**Benefits:**

Error will be raised when running `lightning.tensor` on a unsupported GPU. 

**Possible Drawbacks:**

**Related GitHub Issues:**

https://github.com/PennyLaneAI/pennylane-lightning/issues/799

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 ### Improvements
 
+* Add GPU device compute capability check for Lightning-Tensor
```
Comment (vincentmr) [inline .github/CHANGELOG.md:18]:
```suggestion
* Add GPU device compute capability check for Lightning-Tensor.
```

**Review by vincentmr — APPROVED** [review]
LGTM, cheers @multiphaseCFD .

File: .github/CHANGELOG.md (hunk 1 above)
Comment (multiphaseCFD) [reply .github/CHANGELOG.md:18]: Nice catch!

**Review by LuisAlfredoNu — APPROVED** [review]
good

**Pull request merged**